### PR TITLE
api: Make Derivative Reconstructable

### PR DESCRIPTION
**Pull request by mloubout**
Fixes

https://devitocodes.slack.com/archives/C1HSXP6KB/p1721850267638059

### CODE REVIEW

File: tests/test_derivatives.py [hunk 1]
```diff
@@ -713,6 +713,16 @@ def test_deriv_spec(self):
         assert dxy0.x0 == {y: y+y.spacing/2}
         assert dxy02.x0 == {x: x+x.spacing/2}
 
+    def test_deriv_stagg_plain(self):
```
Comment (EdCaunt) [inline tests/test_derivatives.py:716]: Would it be worth also testing evaluating `staggered=None` onto `staggered=x`? Also potentially mixing `staggered=None` and `staggered=NODE`

**Review by FabioLuporini — APPROVED** [review]
LGTM
